.github/workflows/scripts/upload_reports/upload.py

```diff
@@ -0,0 +1,118 @@
+from argparse import ArgumentParser, Namespace
+import requests
+from pydantic_settings import BaseSettings, SettingsConfigDict
+from pathlib import Path
+import json
+import os
+
+
+class PLOSSSettings(BaseSettings):
+    """Settings for the PennyLane OSS service."""
+
+    model_config = SettingsConfigDict(env_prefix="PENNYLANE_OSS_SERVER_")
+
+    endpoint_url: str
+    """URL of the PennyLane OSS Service."""
+
+    api_key: str
+    """API key to be supplied to the PennyLane OSS Service."""
+
+
+def read_reports() -> dict:
+    """Read the reports from the reports directory."""
+
+    # Look for reports in the absolute path where GitHub Actions writes them
+    github_workspace = os.environ.get("GITHUB_WORKSPACE", "")
+    print(f"GitHub workspace: {github_workspace}")
+    if github_workspace:
+        reports_dir = Path(github_workspace) / ".github" / "test-reports"
+    else:
+        reports_dir = Path(".github/test-reports")
+    print(f"Reports directory: {reports_dir}")
+
+    # Look for XML files in test-report-* subdirectories
+    reports = []
+    for test_report_dir in reports_dir.glob("test-report-*"):
```

Comment:
Does this add a constraint on the naming of files within the test directory? As in, should we assume `test-report` is always pytest autogenerated, and never something we should add ourselves, in case this breaks?

Also, just wondering if [`rglob`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.rglob) would be useful here, as it recursively searches directories?

Reply:
The `test-report-` prefix here is autogenerated and purely for organising the test reports as github artefacts. There shouldn't be any issue with naming conventions, PL devs should never need to be concerned with this when adding unit tests.